File: LogisimPlus/LogisimPlus/logisim/gui/generic/AttrTableModelRow.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.gui.generic
{

	public interface AttrTableModelRow
	{
		string Label {get;}

		string Value {get;set;}

		bool ValueEditable {get;}

		Component getEditor(Window parent);

// JAVA TO C# CONVERTER WARNING: Method 'throws' clauses are not available in C#:
// ORIGINAL LINE: public void setValue(Object value) throws AttrTableSetException;
	}

}

File: logisim/app/src/main/java/logisim/gui/generic/AttrTableModelRow.java
/* Copyright (c) 2011, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.gui.generic;

import java.awt.Component;
import java.awt.Window;

public interface AttrTableModelRow {
	String getLabel();

	String getValue();

	boolean isValueEditable();

	Component getEditor(Window parent);

	void setValue(Object value) throws AttrTableSetException;
}
